src/Categories/Bicategory/Construction/LaxSlice.agda

```diff
@@ -0,0 +1,337 @@
+open import Categories.Category renaming (Category to 1Category)
+open import Categories.Morphism.Reasoning
+
+open import Categories.Bicategory.Extras 𝒞
+
+open import Level
+
+record SliceObj (X : Obj) : Set (t ⊔ o) where
+  constructor sliceobj
+  field
+    {Y} : Obj
+    arr : Y ⇒₁ X
+
+module SliceHom (A : Obj) where
+
+  record Slice⇒₁ (X Y : SliceObj A) : Set (o ⊔ ℓ) where
+    constructor slicearr₁
+    private
+      module X = SliceObj X
+      module Y = SliceObj Y
+
+    field
+      {h} : X.Y ⇒₁ Y.Y
+      Δ   : X.arr ⇒₂ (Y.arr ∘ₕ h)
+
+  record Slice⇒₂ {X Y : SliceObj A} (H K : Slice⇒₁ X Y) : Set (ℓ ⊔ e) where
+    constructor slicearr₂
+    private
+      module Y = SliceObj Y
+      module H = Slice⇒₁ H
+      module K = Slice⇒₁ K
+
+    field
+      {ϕ} : H.h ⇒₂ K.h
+      E   : K.Δ ≈ (Y.arr ▷ ϕ ∘ᵥ H.Δ)
+
+  open hom.Equiv
+  open import Categories.Functor using (Functor)
+
+  _∘'_ : ∀ {X Y : SliceObj A}{H K L : Slice⇒₁ X Y} → Slice⇒₂ K L → Slice⇒₂ H K → Slice⇒₂ H L
+  _∘'_ {X}{Y}{H}{K}{L} (slicearr₂ {ϕ = ϕ} E) (slicearr₂ {ϕ = ψ} F) = slicearr₂ {ϕ = ϕ ∘ᵥ ψ} 
+     (begin
+      L.Δ ≈⟨ E ⟩
+      (Y.arr ▷ ϕ ∘ᵥ K.Δ) ≈⟨ refl⟩∘⟨ F ⟩
+      Y.arr ▷ ϕ ∘ᵥ (Y.arr ▷ ψ ∘ᵥ H.Δ) ≈˘⟨ hom.assoc ⟩
+      (Y.arr ▷ ϕ ∘ᵥ Y.arr ▷ ψ) ∘ᵥ H.Δ ≈⟨ (∘ᵥ-distr-▷ ⟩∘⟨refl) ⟩
+      Y.arr ▷ (ϕ ∘ᵥ ψ) ∘ᵥ H.Δ ∎
+        )
+    where module X = SliceObj X
+          module Y = SliceObj Y
+          module H = Slice⇒₁ H
+          module K = Slice⇒₁ K
+          module L = Slice⇒₁ L
+          open 1Category (hom X.Y A)
+          open HomReasoning
+          open Equiv
+          open import Categories.Morphism.Reasoning (hom X.Y A)
+          open import Relation.Binary.Core using (Rel)
+          open import Function.Base using (_$_)
+          
+  open SliceObj
+  SliceHomCat : SliceObj A → SliceObj A → 1Category (o ⊔ ℓ) (ℓ ⊔ e) e
+  SliceHomCat X Y = record
+    { Obj = Slice⇒₁ X Y
+    ; _⇒_ = Slice⇒₂
+    ; _≈_ = λ where
+      (slicearr₂ {ϕ} _) (slicearr₂ {ψ} _) → ϕ ≈ ψ 
+    ; id = slice-id _
+    ; _∘_ = _∘'_
+    ; assoc = hom.assoc
+    ; sym-assoc = hom.sym-assoc
+    ; identityˡ = hom.identityˡ
+    ; identityʳ = hom.identityʳ
+    ; identity² = hom.identity²
+    ; equiv = record
+      { refl = refl
+      ; sym = sym
+      ; trans = trans }
+    ; ∘-resp-≈ = hom.∘-resp-≈
+    }
+    where
+      slice-id : ∀ (H : Slice⇒₁ X Y) → Slice⇒₂ H H
+      slice-id H = slicearr₂ (begin
+        (H.Δ        ≈˘⟨ identity₂ˡ ⟩
+         id₂ ∘ᵥ H.Δ ≈⟨ (sym ▷id₂ ⟩∘⟨refl) ⟩
+         (arr Y ▷ id₂) ∘ H.Δ
+         ∎))
+        where module X = SliceObj X
+              module H = Slice⇒₁ H
+              open 1Category (hom X.Y A)
+              open HomReasoning
+
+LaxSlice : Obj → Bicategory (o ⊔ ℓ) (ℓ ⊔ e) e (o ⊔ t)
+LaxSlice A   = record
+  { enriched = record
+    { Obj = SliceObj A
+    ; hom = SliceHomCat
+    ; id = const (SliceHom.slicearr₁ ρ⇐)
+    ; ⊚ = _⊚'_
+    ; ⊚-assoc = λ {W X Y Z} →
+      let module W = SliceObj W
```

Comment:
`let` is very badly behaved in agda (it gets inlined), so that tends to make things slow. `where` is preferred. So you should put all of this in a `where` for the record. Or define it as a private function above `LaxSlice`.